games/devtest/mods/unittests/player.lua

```diff
@@ -4,33 +4,40 @@
 local expect = nil
 minetest.register_on_player_hpchange(function(player, hp, reason)
 	if expect == nil then
-		return
+		return hp
 	end
 
 	for key, value in pairs(reason) do
 		assert(expect[key] == value)
 	end
-	for key, value in pairs(expect) do
-		assert(reason[key] == value)
-	end
+    for key, value in pairs(expect) do
+        assert(reason[key] == value)
+    end
 
-	expect = nil
-end)
+    expect = nil
+	
+    if hp == -19 then
+        return -100
```

Comment:
Why put this in on player HP change? Are you trying to test something else, or why can't you just punch with 100 damage (or similar) to begin with?